Habitica/src/main/java/com/habitrpg/android/habitica/ui/adapter/SkillsRecyclerViewAdapter.kt

```diff
@@ -135,10 +136,10 @@ class SkillsRecyclerViewAdapter : RecyclerView.Adapter<SkillsRecyclerViewAdapter
 
         private fun getOwnedCount(key: String): Int {
             return when (key) {
-                "snowball" -> specialItems?.snowball
-                "shinySeed" -> specialItems?.shinySeed
-                "seafoam" -> specialItems?.seafoam
-                "spookySparkles" -> specialItems?.spookySparkles
+                "snowball" -> specialItems?.firstOrNull() { it.key == "snowball" }?.numberOwned
+                "shinySeed" -> specialItems?.firstOrNull() { it.key == "shinySeed" }?.numberOwned
+                "seafoam" -> specialItems?.firstOrNull() { it.key == "seafoam" }?.numberOwned
+                "spookySparkles" -> specialItems?.firstOrNull() { it.key == "spookySparkles" }?.numberOwned

```

Comment:
replace the whole `when` part with just the check. 

```
return specialItems?.firstOrNull() { it.key == key  }?.numberOwned ?: 0
```

this would do the same thing and also handle new keys without needing changes.